home.admin/config.scripts/bonus.lnbits.sh

```diff
@@ -627,24 +622,18 @@ if [ "$1" = "install" ]; then
   echo "# get the github code user(${githubUser}) branch(${tag})"
   sudo rm -r /home/lnbits/lnbits 2>/dev/null
   cd /home/lnbits  || exit 1
-  sudo -u lnbits git clone https://github.com/${githubUser}/lnbits-legend lnbits
+  sudo -u lnbits git clone https://github.com/${githubUser}/lnbits lnbits
   cd /home/lnbits/lnbits || exit 1
   sudo -u lnbits git checkout ${tag} || exit 1
 
   # to the install
   echo "# installing application dependencies"
   cd /home/lnbits/lnbits  || exit 1
 
+  # check if poetry in installed, if not install it
+  sudo -u lnbits which poetry || sudo -u lnbits curl -sSL https://install.python-poetry.org | sudo -u lnbits python3 -
```

Comment:
we use poetry in more apps already namely CLN, backup-plugin and Suez (channel visualization tool). While updateing those had difficulties with poetry being installed for different users as system side and poetry venv packages were mixed in unpredictably. I found that since we have python and pip installed under root we should have poetry too to not interfere with permissions.

so would suggest to use:
```suggestion
  if ! sudo -u lnbits which poetry; then
    echo "# install poetry"
    sudo pip3 install --upgrade pip
    sudo pip3 install poetry
  fi
```

this change doesn't affect the `sudo -u lnbits poetry install` line below.

Reply:
i think its a great change, and checking the binary with `which` is the best thing to fall back to, imho.

Reply:
but still ```curl -sSL https://install.python-poetry.org | python3 -```is the recommended way to install poetry from their website.

Reply:
Yes, this is a problem since the official script install in a  folder not in the PATH in Debian and we need find a way to reliably add it to the PATH for all users (if installed for root) or for the dedicated user (which is even more problematic to test).

example to install for the `bitcoin` user where it is used already:
```
admin@192.168.3.175:~ ₿ curl -sSL https://install.python-poetry.org | sudo -u bitcoin python3 -
Retrieving Poetry metadata

# Welcome to Poetry!

This will download and install the latest version of Poetry,
a dependency and package manager for Python.

It will add the `poetry` command to Poetry's bin directory, located at:

/home/bitcoin/.local/bin

You can uninstall at any time by executing this script with the --uninstall option,
and these changes will be reverted.

Installing Poetry (1.4.0): Done

Poetry (1.4.0) is installed now. Great!

To get started you need Poetry's bin directory (/home/bitcoin/.local/bin) in your `PATH`
environment variable.

Add `export PATH="/home/bitcoin/.local/bin:$PATH"` to your shell configuration file.

Alternatively, you can call Poetry explicitly with `/home/bitcoin/.local/bin/poetry`.

You can test that everything is set up by executing:

`poetry --version`
```
Testing the PATH settings:
```
admin@192.168.3.175:~ ₿ sudo -u bitcoin which poetry; echo $?
1

admin@192.168.3.175:~ ₿ export PATH="/home/bitcoin/.local/bin:$PATH"
admin@192.168.3.175:~ ₿ sudo -u bitcoin which poetry; echo $?
1

admin@192.168.3.175:~ ₿ echo 'export PATH="/home/bitcoin/.local/bin:$PATH"' | sudo tee -a /home/bitcoin/.bashrc
admin@192.168.3.175:~ ₿ sudo -u bitcoin which poetry; echo $?
1
```

Reply:
the only working alternative seems that we would need to use full paths for poetry everywhere